core/src/main/java/io/confluent/kafka/schemaregistry/storage/KafkaStoreReaderThread.java

```diff
@@ -82,54 +77,53 @@ public KafkaStoreReaderThread(ZkUtils zkUtils,
     this.commitInterval = commitInterval;
     this.noopKey = noopKey;
 
-    offsetInSchemasTopic = offsetOfLastConsumedMessage(zkUtils, groupId, topic);
-    log.info("Initialized the consumer offset to " + offsetInSchemasTopic);
     Properties consumerProps = new Properties();
     consumerProps.put("group.id", this.groupId);
     consumerProps.put("client.id", "KafkaStore-reader-" + this.topic);
-    consumerProps.put("zookeeper.connect", kafkaClusterZkUrl);
-    consumerProps.put("auto.offset.reset", "smallest");
+
+    consumerProps.put("bootstrap.servers", bootstrapBrokers);
+    consumerProps.put("auto.offset.reset", "earliest");
     consumerProps.put("auto.commit.enable", "false");
-    consumer = new ZookeeperConsumerConnector(new ConsumerConfig(consumerProps));
-    Map<String, Integer> kafkaStreamConfig = new HashMap<String, Integer>();
-    kafkaStreamConfig.put(topic, 1);
-    Map<String, List<KafkaStream<byte[], byte[]>>> streams =
-        consumer.createMessageStreams(kafkaStreamConfig);
-    List<KafkaStream<byte[], byte[]>> streamsForTheLogTopic = streams.get(topic);
-    // there should be only one kafka partition and hence only one stream
-    if (streamsForTheLogTopic != null && streamsForTheLogTopic.size() != 1) {
+    consumerProps.put("key.deserializer", org.apache.kafka.common.serialization.ByteArrayDeserializer.class);
+    consumerProps.put("value.deserializer", org.apache.kafka.common.serialization.ByteArrayDeserializer.class);
+    this.consumer = new KafkaConsumer<>(consumerProps);
+    this.topicPartition = new TopicPartition(topic, 0); // only one partition.
+    this.consumer.assign(Arrays.asList(this.topicPartition));
+    this.consumer.poll(1); // poll to force communication with the broker before checking if the assignment exists.
+    this.consumer.seekToBeginning(this.topicPartition);
+    if (this.consumer.assignment().size() != 1) {
```

Comment:
On second thought, this is actually kind of weird. We assign the single partition explicitly above, so there's no reason to check if the assignment is different. Perhaps you should call partitionsFor() and verify that there is exactly one partition. You can skip the poll(1) as well.